src/nurse/views.py

```diff
@@ -57,6 +58,22 @@ class NurseViewSet(viewsets.ModelViewSet):
     serializer_class = NurseSerializer
 
 
+class UserOneSignalProfileViewSet(
+    mixins.CreateModelMixin, viewsets.ReadOnlyModelViewSet
+):
+    queryset = UserOneSignalProfile.objects.all()
+    serializer_class = UserOneSignalProfileSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        user = self.request.user
+        return UserOneSignalProfile.objects.filter(user=user)
+
+    def create(self, request, *args, **kwargs):
+        request.data["user"] = request.user.id
+        return super().create(request, *args, **kwargs)
+
```

Comment:
We probably also need something like this so that only the UserOneSignalProfile associated to the request.user is retrieved.
```
    def retrieve(self, request, *args, **kwargs):
        obj = get_object_or_404(self.get_queryset(), user=request.user)
        serializer = self.get_serializer(obj)
        return Response(serializer.data)
```